gcloud/monitoring/group_service_api.py

```diff
@@ -0,0 +1,342 @@
+            service_path=_SERVICE_ADDRESS,
+            port=_DEFAULT_SERVICE_PORT,
+            channel=None,
+            ssl_creds=None,
+            scopes=_ALL_SCOPES,
+            is_idempotent_retrying=True,
+            max_attempts=3,
+            timeout=30):
+        self.defaults = api_callable.ApiCallableDefaults(
+            timeout=timeout,
+            max_attempts=max_attempts,
+            is_idempotent_retrying=is_idempotent_retrying)
+
+        self.stub = config.create_stub(
+            group_service_pb2.beta_create_GroupService_stub,
+            service_path,
+            port,
+            ssl_creds=ssl_creds,
+            channel=channel,
+            scopes=scopes)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        del self.stub
+
+    # Service calls
+    def list_groups(
+            self,
+            name='',
+            children_of_group='',
+            ancestors_of_group='',
+            descendants_of_group='',
+            **kwargs):
+        """
+        Lists the existing groups.
+
+        :type name: string
+        :type children_of_group: string
+        :type ancestors_of_group: string
+        :type descendants_of_group: string
+        """
+
+        list_groups_request = group_service_pb2.ListGroupsRequest(
+            name=name,
+            children_of_group=children_of_group,
+            ancestors_of_group=ancestors_of_group,
+            descendants_of_group=descendants_of_group,
+            **kwargs)
+        return self.list_groups_callable()(list_groups_request)
+
+    def list_groups_callable(
+            self,
+            is_retrying=None,
+            max_attempts=None,
+            page_streaming=_LIST_GROUPS_DESCRIPTOR):
+        return api_callable.idempotent_callable(
+            self.stub.ListGroups,
+            page_streaming=page_streaming,
+            is_retrying=is_retrying,
+            max_attempts=max_attempts,
+            defaults=self.defaults)
+
+    def list_groups_for_resource(
+            self,
+            name='',
+            monitored_resource=None,
+            **kwargs):
+        """
+        Lists the groups containing a specific monitored resource.
+
+        :type name: string
+        :type monitored_resource: monitored_resource_pb2.MonitoredResource
+        """
+        if monitored_resource is None:
+            monitored_resource = monitored_resource_pb2.MonitoredResource()
+        list_groups_for_resource_request = group_service_pb2.ListGroupsForResourceRequest(
+            name=name,
+            monitored_resource=monitored_resource,
+            **kwargs)
+        return self.list_groups_for_resource_callable()(list_groups_for_resource_request)
+
+    def list_groups_for_resource_callable(
+            self,
+            is_retrying=None,
+            max_attempts=None,
+            page_streaming=_LIST_GROUPS_FOR_RESOURCE_DESCRIPTOR):
```

Comment:
No docstring?

Also, I get why you need an 8-space indent here, but you wouldn't (I think?) if there was a docstring.